File: src/FactoryMethod/src/Midia.java
import java.util.ArrayList;

public abstract class Midia {
    protected String titulo;
    protected String descricao;
    protected int anoLancamento;
    protected int duracao;
    protected String genero;
    protected String classificacao;
    protected ArrayList<String> elenco;

    public Midia(String titulo, String descricao, int anoLancamento, int duracao, String genero, String classificacao, ArrayList<String> elenco) {
        this.titulo = titulo;
        this.descricao = descricao;
        this.anoLancamento = anoLancamento;
        this.duracao = duracao;
        this.genero = genero;
        this.classificacao = classificacao;
        this.elenco = elenco != null ? elenco : new ArrayList<>();
    }

    // Construtor e métodos abstratos ou concretos, conforme necessário
}

File: src/FactoryMethod/src/models/midia.ts
export abstract class Midia {
    protected titulo: string;
    protected descricao: string;
    protected anoLancamento: number;
    protected duracao: number;
    protected genero: string;
    protected classificacao: string;
    protected elenco: string[];

    constructor(titulo: string, descricao: string, anoLancamento: number, duracao: number, genero: string, classificacao: string, elenco: string[]) {
        this.titulo = titulo;
        this.descricao = descricao;
        this.anoLancamento = anoLancamento;
        this.duracao = duracao;
        this.genero = genero;
        this.classificacao = classificacao;
        this.elenco = elenco ?? [];
    }
}